chingu-voyages/v52-tier3-team-38
# Gary/feature/address validation

# Description
This PR handles finalizes address validation. Addresses entered through the submit request form are checked in the LA dataset to ensure they exist. I made some adjustments such as adding phone number and address coordinates to simplify whats to come and better align with requirements.

## Related to the following Jira/Trello issue 
https://trello.com/c/RHkbyOL3/22-implement-data-validation-to-ensure-addresses-match-the-provided-dataset-from-los-angeles-addresses

## Main changes explained:
- bookAppointment api was refactored to include address validation to ensure the address entered by a user is valid.
- bookAppointment api now inserts latitude and longitude coordinates of addresses for google maps api use. 
- The submit request appointment form requires the submission of phone numbers to better align with requirements.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. Go to the submit request form and test entering your info to schedule an appointment.
4. Make the form fail by entering an address NOT listed in the suggestions. You should see an error regarding this upon submission.
5. Try to submit a request by using a correct address (any that is listed in the suggestions provided)
6. Verify the appointment info was saved to the data with the new information, including coordinates of address (lat and lon), and phone number.

## Note:
I do not believe the user details is needed with these new changes. I would like to discuss removing them to clean up the database and simplify some processes.
